### PR TITLE
PEP8 changes

### PR DESCRIPTION
**Pull request by Hazboun6**
These changes clean up the code for linting, but do not go as far as PR #137 .  I used `autopep8` to edit whitespace but that was mostly indenting. It did not change equation spacing for instance.

### CODE REVIEW

**Comment by Hazboun6** [comment]
@kdolum curious what you think about this PR.

**Comment by kdolum** [comment]
This still seems to be formatting things the way it wants without giving us control.  For example in the diffs for `chromatic.py`, line 172 (old) 174 (new) it put spaces around slash and minus sign.  In `README.md`, it really left the rails, changing "power-law" to "power - law" in text.

File: enterprise_extensions/chromatic/README.md [hunk 1]
```diff
@@ -12,14 +12,16 @@ n_earth = SW.ACE_SWEPAM_Parameter()('n_earth')
 sw = SW.solar_wind(n_earth=n_earth)
 mean_sw = deterministic_signals.Deterministic(sw, name='mean_sw')
 ```
-3. There is also a function to create a Fourier design matrix for constructing a Gaussian process. This can be used as desired, but works best as a *perturbation* to the mean signal defined above.
```
Comment (paulthebaker) [inline enterprise_extensions/chromatic/README.md:15]: We probably don't need to enforce style in a README.  If we manually undo this change, will it cause a lint error?

**Review by paulthebaker — APPROVED** [review]
I have a very specific and hard to articulate "space between operator rule" that disagrees with PEP8.  PEP8-ifiying the code at least makes everything consistent, and it wasn't before.  I still think there are cases where the reformatted code is harder to read, but I'm okay signing off on this.

There are a few cleanup changes that I'd like to see. But they are not necessary right now, if it's too much trouble.  There are a few cases where constants that are defined in `enterprise.constants` are not used in `e_e`.  I found 86400 (`const.day`) and yr_in_sec (`const.yr`) a bunch.  There's also the number `4.148808e3` which is used in the solar wind code.  It might be helpful to assign it to a variable with a comment saying where it comes from.

**Comment by paulthebaker** [comment]
@kdolum @Hazboun6, I think we should still turn off the "space around operator rule" in `flake8`, `E225`.  Does the auto formatter do it's own thing or does it follow `flake8`?

File: enterprise_extensions/chromatic/README.md (hunk 1 above)
Comment (Hazboun6) [reply enterprise_extensions/chromatic/README.md:15]: I don't think it will. I think I was a bit draconian with `autopep8`. `flake8` should only actually check `.py` files.

**Comment by Hazboun6** [comment]
Okay, @kdolum and @paulthebaker , I agree. We can turn off E225 in our `.flake8` config file and it will ignore this error. The nice thing is that `flake8` will help us configure the rules we do want to follow. 

Also, to be clear these changes were something I did locally with the specified tools **once**. They wouldn't be run automatically, and we wouldn't assume they would be run by other devs. (That said, we could add them to pre-commit tools, but maybe we can save that for another release. )

The `autopep8` tool **will ignore rules**, but I obviously didn't have it ignore `E225` . So I would probably just start over again as that was the first commit in this long string of changes 🙃 .  That took me about an hour yesterday, so it might take a bit to get it done again.

**Comment by Hazboun6** [comment]
Sorry,  I didn't fully absorb your full message above. 
> That was the big issue with `black`, it's not configurable. If the new one Jeff is using can be configured, it should be an easy fix.
Agreed! And yes it is configurable, in that it can ignore various PEP8 rules. Not sure it is more than that. Again though, let's put off using it as a tool for everyone and rather just use it once for now and get it into the master branch after we get these fixes into the hot fix. 

> The state of things at the commit "various PEP8 flake8 fixes" `aeb51e7` does not have the reformatted equation spacing. After "add lint changes from jellis18" `19732d3` all of the equation spacing is messed up. 
That is weird. I'll go investigate and see how far I can rewind. 

> You can probably rewind back to the earlier commit. The [test failure](https://github.com/nanograv/enterprise_extensions/runs/3749539381) at the earlier commit was `black`. We want to completely remove `black` from our testing and config.

Yeah, I commented it out for now, but I'll just remove it.

**Comment by kdolum** [comment]
I have no code in `e_e`, so I probably don't deserve much say in how it's managed.  So here's my opinion, but feel free to ignore it.

The original impetus for these changes seemed to be to have some automated tool that would run on proposed modifications to find bugs like missing imports and perhaps stylistic problems like variables that are set but never used.  This is what I think of as "linting".  I think this is a good idea.  I don't see why anyone would object to it.

But this then became entangled with standardization of formatting, involving tools that automatically reformat your code and those that signal errors when your code is not formatted the way that they prefer.  I think this is a separate issue.  Surely we can check for bugs without insisting on a standard format.

Personally, I'm against autoformatting or rejecting changes that don't conform to fixed formatting standards.  I'm not against a standard format that we encourage people to use.  It's fine with me, for example, to decide that comments should be lined up or they should have some fixed number of spaces or whatever.  But I think a wise person will sometimes break such formatting.  For example, I think we should have a standard maximum line width,but sometimes you just have code that will be more difficult to understand if you break the line than if you keep it together,even though it may be wrapped when it is displayed.  I think in such cases people should be permitted to write the code the way they feel is best, with the system neither reformatting the code for them nor refusing to allow the code the way they have written it.

So I would favor automatic checking for bugs, no reformatting and no rejection of nonconforming formats.  I've no complaint with an automated process that would issue analyses saying "by the way your code is nonconforming", only with automatic rejection.

**Comment by paulthebaker** [comment]
@kdolum, `flake8` is the linter we use in `enterprise`.  We used to have a bunch of rules turned off, either globally or for specific files. Justin adding `black` to the tests is when things got weird over there.  Using `flake8` we can select which rules we want to enforce.  And we can always add per file or even per line exceptions if needed.

It's just a matter of picking the minimal set of rules we want globally.

**Comment by Hazboun6** [comment]
@kdolum Strong opinions make for useful conversions, at least my inner New York self believes that :-)

> But this then became entangled with standardization of formatting, involving tools that automatically reformat your code and those that signal errors when your code is not formatted the way that they prefer. I think this is a separate issue. Surely we can check for bugs without insisting on a standard format.

This is just an issue of the easily available tools. `flake8` tests **both**, and one can ignore whatever `PEP8` errors we want to ignore. These are the standard tools that are available. Which doesn't mean we should do it, but it does just make it easier.

**Comment by Hazboun6** [comment]
> The state of things at the commit "various PEP8 flake8 fixes" aeb51e7 does not have the reformatted equation spacing. After "add lint changes from jellis18" 19732d3 all of the equation spacing is messed up.

@paulthebaker it does look like I made the `autopep8` changes early on, unless you are seeing things differently. I can't imagine a clean way to go without just starting over...

**Comment by paulthebaker** [comment]
There's not a way to rewrite history, so you may want to start over... `git revert 19732d3 4bdd09f` will apply the "anti-diffs" which can get you to the right spot, but would have a record of all of the commits.

**Comment by Hazboun6** [comment]
>There are a few cleanup changes that I'd like to see. But they are not necessary right now, if it's too much trouble. There are a few cases where constants that are defined in enterprise.constants are not used in e_e. I found 86400 (const.day) and yr_in_sec (const.yr) a bunch. There's also the number 4.148808e3 which is used in the solar wind code. It might be helpful to assign it to a variable with a comment saying where it comes from.

I see a few of these, but I think we should push through the various whitespace changes and clean that up later.

**Comment by Hazboun6** [comment]
I've closed this PR in favor of PR #140 .
